### PR TITLE
feat: migrate social image card to `gatsby-plugin-image`

### PR DESCRIPTION
**Pull request by feedm3**
We had some last remainings of the outdated `gatsby-image`, which I migrated to `gatsby-plugin-image`. 

- We are now 100% up-to-date
- This also resolved the last warning we had in our build logs
- We can now use the new [Gatsby Image CDN](https://www.gatsbyjs.com/products/cloud/image-cdn/), which should speed up our build on Gatsby Cloud very much 💪 

I will [enable the Image CDN](https://support.gatsbyjs.com/hc/en-us/articles/4426393233171) once we merged this PR.

### CODE REVIEW

File: package.json [hunk 1]
```diff
@@ -43,9 +43,8 @@
     "canvas": "^2.9.0",
     "date-fns": "^2.28.0",
     "fast-xml-parser": "^4.0.6",
-    "gatsby": "4.11.2",
+    "gatsby": "4.11.3",
     "gatsby-background-image": "^1.6.0",
-    "gatsby-image": "^3.11.0",
```
Comment (feedm3) [inline package.json:48]: Bye bye ✋

File: src/templates/career-details.tsx [hunk 2]
```diff
@@ -24,7 +24,9 @@ interface CareerPageProps {
 const CareerPage = (props: CareerPageProps): JSX.Element => {
   const { pageContext } = props;
   const position = props.data.syPersonioJob;
-  const socialCardPath = position.socialCardFile.childImageSharp.fixed.src;
+  const socialCardPath =
```
Comment (feedm3) [inline src/templates/career-details.tsx:27]: Still works: https://www.heymeta.com/url/satellytescommain21751-choreimage.gtsb.io/career/staff-software-engineer/

Screenshot 2022-04-12 at 08 56 09

File: package.json (hunk 1 above)
Comment (georgiee) [reply package.json:48]: byee 👋

File: package.json [hunk 3]
```diff
@@ -72,7 +71,7 @@
     "gatsby-transformer-remark": "^5.11.1",
     "gatsby-transformer-sharp": "^4.11.0",
     "html-entities": "^2.3.2",
-    "i18next": "^21.6.15",
+    "i18next": "^21.6.16",
```
Comment (georgiee) [inline package.json:74]: so I guess you smuggle in some dependency upgrade to save some time next Monday? 😄

File: src/types.ts [hunk 4]
```diff
@@ -38,7 +38,11 @@ export interface SyPersonioJob {
   sections: SyPersonioJobSection[];
 
   // added via `onCreateNode`
-  socialCardFile: PlainFixedImageSharpSource;
+  socialCardFile: {
```
Comment (georgiee) [inline src/types.ts:41]: 👍

**Review by georgiee — APPROVED** [review]
Let's merge that, thanks for thinking of that 🤩

File: package.json (hunk 3 above)
Comment (feedm3) [reply package.json:74]: guilty 😄

**Pull request merged**